hoot-core/src/main/cpp/hoot/core/OsmMap.cpp

```diff
@@ -189,6 +189,37 @@ void OsmMap::addNode(const NodePtr& n)
   //_nodeCounter = std::min(n->getId() - 1, _nodeCounter);
 }
 
+void OsmMap::addNodes(const std::vector<NodePtr>& nodes)
+{
+  if (nodes.size() > 0)
+  {
+    long minId = nodes[0]->getId();
+    long maxId = minId;
+
+    // this seemed like a clever optimization. However, this impacts the BigPertyCmd.sh test b/c
+    // it modifies the order in which the elements are written to the output. Which presumably (?)
+    // impacts the ID when reading the file with re-numbering. Sad.
+//    size_t minBuckets = _nodes.size() + nodes.size() * 1.1;
+//    if (_nodes.bucket_count() < minBuckets)
+//    {
+//      _nodes.resize(minBuckets);
+//    }
```

Comment:
It doesn't cause a crash. The results still appear to be valid, but we're writing them out in a different order which causes the elements to have different IDs when they go through Hadoop. I suppose I could update the test, but I didn't want to risk the fallout of other regression tests breaking at this time.